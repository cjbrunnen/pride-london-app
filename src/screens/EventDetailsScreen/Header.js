// @flow
import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  StatusBar
} from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import { eventDetailsHeaderBgColor } from "../../constants/colors";

type HeaderProps = {
  onBackButtonPress: () => void
};

const Header = ({ onBackButtonPress }: HeaderProps) => (
  <SafeAreaView style={styles.container}>
    <StatusBar barStyle="default" animated />
    <View style={styles.header}>
      <TouchableOpacity onPress={onBackButtonPress}>
        <Text>Back</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: eventDetailsHeaderBgColor
  },
  header: {
    height: 180,
    padding: 22
  }
});

export default Header;
