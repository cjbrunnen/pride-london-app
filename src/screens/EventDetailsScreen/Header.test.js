// @flow
import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

it("renders correctly", () => {
  const output = shallow(<Header onBackButtonPress={() => {}} />);
  expect(output).toMatchSnapshot();
});

it("calls callback when pressed", () => {
  const onPress = jest.fn();
  const output = shallow(<Header onBackButtonPress={onPress} />);

  const touchable = output.find("TouchableOpacity");
  touchable.simulate("press");

  expect(onPress).toHaveBeenCalled();
});
