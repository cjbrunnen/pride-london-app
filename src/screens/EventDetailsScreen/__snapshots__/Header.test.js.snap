// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withOrientation
  style={
    Object {
      "backgroundColor": "#F2F2F2",
    }
  }
>
  <StatusBar
    animated={true}
    barStyle="default"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "height": 180,
        "padding": 22,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Back
      </Text>
    </TouchableOpacity>
  </View>
</withOrientation>
`;
